tools/crm_mon.c

```diff
@@ -380,7 +380,8 @@ static struct crm_option long_options[] = {
     {"tickets",        0, 0, 'c', "\t\tDisplay cluster tickets"},
     {"watch-fencing",  0, 0, 'W', "\tListen for fencing events. For use with --external-agent"},
     {"fence-history",  2, 0, 'm', "Show fence history\n"
-                                  "\t\t\t\t\t0=off, 1=failures, 2=add successes and pending (default without value),\n"
+                                  "\t\t\t\t\t0=off, 1=failures and pending (default without option),\n"
+                                  "\t\t\t\t\t2=add successes (default without value for option),\n"
```

Comment:
I'm fine with this approach, but what do you think about just requiring an argument, to avoid the option parser oddness?

Reply:
Well, thought about that of course.
On the other hand I like the idea that 99% of all cases will be covered with the default (failures and now pending) or '-m' and the value will just be needed in very special cases from my current pov.
And it is not the only occurrence of optional values. Didn't count them in the pacemaker-tools (unfortunately optional_argument is just used in the library and not where it is used).
And it is the behavior of getopts.